aspnet/security/accconfirm.rst

```diff
@@ -1,236 +1,209 @@
+
+You generally want to prevent new users from posting any data to your web site before they have been confirmed by email, a SMS text message or another mechanism. In the sections below, we will enable email confirmation and modify the code to prevent newly registered  users from logging in until their email has been confirmed.
 
 Setup up Email provider
------------------------
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We'll use the :ref:`Options pattern <options-config-objects>` to access the user account and key settings. For more information, see :doc:`../fundamentals/configuration`.
 
-Although this tutorial only shows how to add email notification through `SendGrid <https://sendgrid.com/>`_, you can send email using SMTP and other mechanisms.
- - Install SendGrid NuGet package
+ - Create a class to fetch the secure email key. For this sample, the ``AuthMessageSenderOptions`` class is created in the *Services/AuthMessageSenderOptions.cs* file.
+
+  .. literalinclude:: accconfirm/sample/WebApplication1/src/WebApplication1/Services/AuthMessageSenderOptions.cs
+   :language: c#
+   :lines: 3-7
+   :dedent: 4
+
+Set the ``SendGridUser`` and ``SendGridKey`` with the `secret manager tool <http://docs.asp.net/en/latest/security/app-secrets.html>`_.
+
+Configure startup to use ``AuthMessageSenderOptions``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Add ``AuthMessageSenderOptions`` to the service container at the end of the ``ConfigureServices`` method in the *Startup.cs* file:
+
+.. literalinclude:: accconfirm/sample/WebApplication1/src/WebApplication1/Startup.cs
+   :language: c#
+   :lines: 81-85
+   :emphasize-lines: 4
+   :dedent: 8
+
+.. ToDo figure out bolding in next line
+
+Configure the ``AuthMessageSender`` class
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This tutorial shows how to add email notification through `SendGrid <https://sendgrid.com/>`_, but you can send email using SMTP and other mechanisms.
+ - Install the SendGrid NuGet package. From the Package Manager Console,  enter the following the following command:
+``Install-Package SendGrid``
  - Go to the `Azure SendGrid sign up page <http://azure.microsoft.com/en-us/marketplace/partners/sendgrid/sendgrid-azure/>`_ and register for a free SendGrid account.
- - Add code in **MessageServices.cs** similar to the following to configure SendGrid
+ - Add code in *Services/MessageServices.cs* similar to the following to configure SendGrid
 
-.. code-block:: c#
+.. literalinclude:: accconfirm/sample/WebApplication1/src/WebApplication1/Services/MessageServices.cs
+   :language: c#
+   :lines: 12-51
+   :dedent: 4 
```

Comment:
That's outside the namespace and routinely handled with the quick fix button in VS.